src/Raven.Server/Documents/Handlers/DocumentHandler.cs

```diff
@@ -236,13 +238,39 @@ private async Task GetDocumentsByIdAsync(DocumentsOperationContext context, Micr
                     {
                         document = Database.DocumentsStorage.Get(context, id);
                     }
-
-                    if (document == null && ids.Count == 1)
+                    
+                    if (document == null)
                     {
-                        HttpContext.Response.StatusCode = GetStringFromHeaders(Constants.Headers.IfNoneMatch) == HttpCache.NotFoundResponse
-                        ? (int)HttpStatusCode.NotModified
-                        : (int)HttpStatusCode.NotFound;
-                        return;
+                        if (loadFromClusterWideTx)
+                        {
+                            Debug.Assert(includeCompareExchangeValues != null, nameof(includeCompareExchangeValues) + " != null");
+                            includeCompareExchangeValues.AddDocument(ClusterTransactionCommand.GetAtomicGuardKey(id));
+                            continue;
+                        }
+                        if (ids.Count == 1)
+                        {
+                            HttpContext.Response.StatusCode = GetStringFromHeaders(Constants.Headers.IfNoneMatch) == HttpCache.NotFoundResponse
+                                ? (int)HttpStatusCode.NotModified
+                                : (int)HttpStatusCode.NotFound;
+                            return;
+                        }
+                    }
+                    else if (loadFromClusterWideTx && 
+                             document.ChangeVector.Contains(ChangeVectorParser.TrxnTag) == false)
```

Comment:
better to check the `ClusterTransactionId`. the `TrxnTag` can be from external replication

Reply:
Good point